make static: `mypy --no-site-packages` pass reports 37 spurious untyped-decorator errors in lsp/

## Observation

The `make static` target runs **two** mypy passes:

```make
static:
	$(PYTHON) -m ruff check .
	$(PYTHON) -m mypy .
	$(PYTHON) -m mypy . --no-site-packages
```

The first mypy pass is clean (`Success: no issues found in 50 source files`). The second — `mypy . --no-site-packages` — reports **37 errors** in two files:

- `lsp/mcp_server.py`: 18 errors, e.g.
  ```
  lsp/mcp_server.py:175: error: Untyped decorator makes function "check_file" untyped  [untyped-decorator]
  lsp/mcp_server.py:191: error: Untyped decorator makes function "goal_at" untyped  [untyped-decorator]
  ...
  ```
- `lsp/lsp_server.py`: 19 errors of the same shape (`on_did_open`, `on_did_save`, etc.).

Every error is an `untyped-decorator` complaint because the decorators come from `mcp.Server` / `pygls.Server`, which aren't on mypy's path when site-packages is excluded.

## CI doesn't see these

The CI `static` job ([`.github/workflows/test_deduce.yml`](.github/workflows/test_deduce.yml)) only runs `mypy .` (with site-packages, after `pip install -r requirements-lsp.txt`), so this noise is **local-only**. As a contributor running `make static` before pushing, the wall of 37 errors masks any real signal from the clean first pass.

## Why it exists

The `--no-site-packages` pass was added to keep type-checking honest when contributors haven't installed optional deps. Precedent: [`be60b08`](https://github.com/jsiek/deduce/commit/be60b08) ("Make parser.py Meta() ignore tolerate --no-site-packages mypy") fixed an analogous problem in `parser.py` by tweaking the `# type: ignore` codes so both passes stay clean.

## Options to fix

1. **Per-decorator suppression on the LSP/MCP files.** Add `# type: ignore[untyped-decorator]` (plus `unused-ignore` in the same comment, so the site-packages pass doesn't trip `warn_unused_ignores`). Same trick as `be60b08`.
2. **Module-level exclusion.** Add `lsp/lsp_server.py` and `lsp/mcp_server.py` to a `[mypy-...]` block in `pyproject.toml` that disables `untyped-decorator` — or use `# mypy: disable-error-code=\"untyped-decorator\"` at the top of each file.
3. **Drop the `--no-site-packages` pass.** It's not run in CI; if no one needs it, removing the second line in the `static` Makefile target eliminates the noise entirely. (Cheapest, but loses the no-deps sanity check that `be60b08`'s scope implies someone cared about.)

Option 1 matches existing precedent and keeps `make static` aligned with CI. Happy to send a PR.

— Filed during a new-user UX walk-through after `make static` produced 37 errors immediately after a one-line `examples/*.pf` change.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -75,21 +75,21 @@ strict_equality = true
 disallow_subclassing_any = true
 # disallow_untyped_decorators is enabled globally now that the CI
 # workflow installs requirements-lsp.txt (`pygls` and `mcp`). Both
-# ship inline type annotations, so the LSP/MCP decorators no longer
-# resolve to Any in CI. Developers who skip the LSP install will see
-# this flag fire on `lsp/lsp_server.py` / `lsp/mcp_server.py`; install
-# `requirements-lsp.txt` to develop on those modules.
+# ship inline type annotations, so the LSP/MCP decorators resolve to
+# real types in the normal `mypy .` pass. The local `--no-site-packages`
+# pass intentionally hides those packages; the LSP/MCP module override
+# below suppresses only that decorator-boundary noise.
 disallow_untyped_decorators = true
 no_implicit_reexport = true
 exclude = [
     '^test/',
-    '^test-deduce\.py$',
+    '^test-deduce\\.py$',
     '^tmp/',
     '^lark/',
     '^editor/',
     '^live_code_vercel_api/',
     '^autograder_docker/',
-    '^\.claude/',
+    '^\\.claude/',
     '^gh_pages/',
 ]
 
@@ -154,6 +154,14 @@ disallow_untyped_decorators = true
 disallow_incomplete_defs = true
 check_untyped_defs = true
 warn_return_any = true
+
+# Keep the no-dependencies mypy pass useful without weakening the rest
+# of strict checking for the LSP/MCP adapters. When `--no-site-packages`
+# hides pygls/mcp, their decorators become Any and otherwise produce a
+# wall of untyped-decorator errors.
+[[tool.mypy.overrides]]
+module = ["lsp.lsp_server", "lsp.mcp_server"]
+disable_error_code = ["untyped-decorator"]
 
 # Submodules of abstract_syntax re-export sibling names through the
 # compatibility facade in abstract_syntax/__init__.py, which conflicts

